File: DotNet/Shared/Application/Models/Kafka/ResourceKey.cs
using System.Text.Json.Serialization;

namespace LantanaGroup.Link.Shared.Application.Models.Kafka;

public class ResourceKey
{
    [JsonPropertyName("facilityId")]
    public string FacilityId { get; set; } = string.Empty;
    [JsonPropertyName("patientId")]
    public string PatientId { get; set; } = string.Empty;
}

File: Java/shared/src/main/java/com/lantanagroup/link/shared/kafka/records/ResourceKey.java
package com.lantanagroup.link.shared.kafka.records;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ResourceKey {
    private String facilityId;
    private String patientId;
}
